corral/util/log.pony

```diff
@@ -99,16 +19,27 @@ primitive CodeLogFormatter is LogFormatter
      .> append(": ")
      .> append(msg)
 
-primitive LevelLogFormatter is LogFormatter
-  fun apply(level: LogLevel, msg: String, loc: SourceLoc): String =>
-    (recover String(level.string().size()
-      + msg.size()
-      + 4)
-    end)
-     .> append(level.string())
-     .> append(": ")
-     .> append(msg)
+primitive LevelLogFormatter
+  fun apply(level: LogLevel): LogFormatter =>
```

Comment:
this seems odd. it only takes a log level?

How is this being used? The lambda concerns me. I'm not sure why that is needed.

Reply:
If I understand the original code properly, the previous LevelLogFormatter prepended the log type (fine, info, warn, error) to the message being logged. It is only used in main.pony, either as a one-off before exiting or after parsing the command line arguments.

If you like, I could get rid of it by changing the call sites to add the log message type.

Reply:
Ah so the existing pony standard library logger doesn't have the log level. That makes sense. So, this wraps every log call later and appends the level. That makes sense.

Can you add some doc comments to both the apply method and the primitive to make clear what it is doing?